vernac/vernacextend.mli

```diff
@@ -156,14 +156,36 @@ type (_, _) ty_sig =
 
 type ty_ml = TyML : bool (* deprecated *) * ('r, 's) ty_sig * 'r * 's option -> ty_ml
 
-(** Wrapper to dynamically extend vernacular commands. *)
+(** Wrapper to dynamically extend vernacular commands.
+
+    This is used by coqpp VERNAC EXTEND.
+
+    Commands added by plugins at Declare ML Module / Require time should provide [plugin].
+
+    Commands added without providing [plugin] cannot be removed from
+    the grammar or modified. Not passing [plugin] is possible for
+    non-plugin coq-core commands and deprecated for all other callers.
+*)
 val vernac_extend :
   ?plugin:string ->
```

Comment:
Maybe I'd prefer something like
```ocaml
source:[ `CoqBuiltin | `Plugin of string | `ExternalBuiltin (* deprecated *) ] ->
```
so that we can actually print a deprecation warning and eventually remove it.

Reply:
I mean, forgetting to pass an optional argument seems too fragile to signal an intention

Reply:
Maybe it should be "only coqpp is supposed to call this command" and "having coqpp files without a DECLARE PLUGIN outside the statically linked code is deprecated"

Reply:
>I mean, not passing an optional argument seems too fragile to signal an intention (ie declare a Coq builtin)

We can add `DECLARE NOT A PLUGIN` to coqpp and change the API to `plugin: string option` if we want to be more explicit.